lib/cocoapods-core/source/aggregate.rb

```diff
@@ -187,8 +165,19 @@ def update_search_index(search_data)
       # @!group Private helpers
       #-----------------------------------------------------------------------#
 
-      # Returns the search related information from the most representative
-      # specification of the set following keys:
+      # Generates search data for given array of sets.
+      def generate_search_index_for_sets(sets)
+        result = {}
+        sets.each do |set|
+          word_list_from_set(set).each do |w|
+            (result[w] ||= []).push(set.name)
```

Comment:
Depending on the average number of words in a wordlist, using the set name/identifier as a symbol will use a bit less space. If it's only used to generate the file index, and not for storing it in memory (and search operations) it's not that crucial.